benchmark/FindCXXStandard.cmake

```diff
@@ -16,34 +16,42 @@ file(WRITE "${SOURCE_FILE}"
     "int main() { }                                                 \n"
 )
 
-# Check for C++14
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+message(STATUS "Check for C++14")
 try_compile(_supports_14 ${CMAKE_BINARY_DIR} ${SOURCE_FILE}
     CMAKE_FLAGS -DCMAKE_CXX_STANDARD=14
     COMPILE_DEFINITIONS -DSTANDARD_DATE=201402L
 )
 if (_supports_14)
     set(CXX_STANDARD 2014)
+    set(CMAKE_CXX_STANDARD 14)
```

Comment:
I'd rather keep setting `CXX_STANDARD`. We don't want to overwrite `CMAKE_CXX_STANDARD`, which is used for something different.